newm-server/src/main/kotlin/io/newm/server/features/song/repo/SongRepositoryImpl.kt

```diff
@@ -330,8 +333,8 @@ internal class SongRepositoryImpl(
             throw HttpUnprocessableEntityException("streamUrl is null")
         }
 
-        val songStreamUrl = URL(song.streamUrl)
-        val mediaHostUrl = URL(environment.getConfigString("aws.cloudFront.audioStream.hostUrl"))
+        val songStreamUrl = URI.create(song.streamUrl).toURL()
+        val mediaHostUrl = URI.create(environment.getConfigString("aws.cloudFront.audioStream.hostUrl")).toURL()
```

Comment:
Why the URI.create()?

Reply:
@wlara URL(song.streamUrl) is deprecated.

![image](https://github.com/projectNEWM/newm-server/assets/245918/aeb71a68-4734-40c8-8f86-098378cff226)